### PR TITLE
fix: metadata tags capturing in v0 transformation

### PR DESCRIPTION
**Pull request by Jayachand**
## What are the changes introduced in this PR?

Write a brief explainer on your code changes.

## What is the related Linear task?

fixes DAT-1355

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3492/test-report.html

**Review by kanishkkatara — APPROVED** [review]

**Comment by abhimanyubabbar** [comment]
@Jayachand : Can you add brief description and a ticket for traceability purposes as well ?

File: src/util/customTransformer.js [hunk 1]
```diff
@@ -253,8 +253,10 @@ async function runUserTransform(
     const tags = {
       identifier: 'v0',
       errored: transformationError ? true : false,
-      ...(events.length && events[0].metadata ? getMetadata(events[0].metadata) : {}),
-      ...(events.length && events[0].metadata ? getTransformationMetadata(events[0].metadata) : {}),
+      ...(Object.keys(eventsMetadata).length ? getMetadata(Object.values(eventsMetadata)[0]) : {}),
+      ...(Object.keys(eventsMetadata).length
+        ? getTransformationMetadata(Object.values(eventsMetadata)[0])
+        : {}),
```
Comment (abhimanyubabbar) [inline src/util/customTransformer.js:259]: Can you explain the issue here a bit ?

Comment (Jayachand) [reply src/util/customTransformer.js:259]: v0 execute function receives only event messages. metadata is not attached to them. Earlier tag extraction `getMetadata(events[0].metdata)` always returns nil. So changed extraction to eventsMetadata

**Review by abhimanyubabbar — APPROVED** [review]

**Pull request merged**